core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftTransportPool.java

```diff
@@ -76,7 +76,8 @@ private ThriftTransportPool(LongSupplier maxAgeMillis) {
         final long minNanos = MILLISECONDS.toNanos(250);
         final long maxNanos = MINUTES.toNanos(1);
         long lastRun = System.nanoTime();
-        while (!connectionPool.shutdown) {
+        // loop often, to detect shutdowns quickly
+        while (!connectionPool.awaitShutdown(250)) {
```

Comment:
good point. CountDownLatch definitely simplifies the wait/wake.
I didn't replace the `shutdown` variable yet. it's checked in other places without waiting. `CountDownLatch.getCount` is documented as "This method is typically used for debugging and testing purposes.", so was hesitant. Should I?

Reply:
`CountDownLatch.getCount` is returning a `volatile int` and the `shutdown` variable is a `volatile boolean`. There may be a little bit of a performance penalty for calling `CountDownLatch.getCount` over checking `shutdown`, but I think that may be offset by reducing the compexity of have to maintain the state of two objects. I would use `CountDownLatch.getCount` in this case. The javadoc suggests what it could be used for, it doesn't say not to use it.

Reply:
removed the `shutdown` field